### PR TITLE
initial credential support

### PR DESCRIPTION
**Pull request by 37ng**
Adds `Credential` struct with proto conversions.
A few tidy-up changes.

### CODE REVIEW

**Comment by 37ng** [comment]
* **#687** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/687?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @37ng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: xmtp_id/src/lib.rs [hunk 1]
```diff
@@ -43,19 +51,50 @@ impl InboxOwner for LocalWallet {
...
 }
 
 impl Identity {
     /// Generate a new, empty ID for an account address.
     /// A nonce is used to ensure uniqueness of the ID.
-    pub fn new(address: String) -> Self {
+    pub fn new(
+        address: String,
+        nonce: &u64,
+        credential: Credential,
```
Comment (neekolas) [inline xmtp_id/src/lib.rs:88]: It seems like we have everything needed to generate the credential from the address and nonce

Comment (37ng) [reply xmtp_id/src/lib.rs:88]: Makes sense. I added a `create_credential()` func, I think that's needed to interface with open mls?

**Comment by 37ng** [comment]
replaced by #688

**Pull request closed without merging**